### PR TITLE
Feature CSS handle

### PR DESCRIPTION
**Pull request by klzns**
#### What problem is this solving?

Add new CSS Handle called `modal` so it's possible to change its `z-index` if needed.

Also:
- migrate to css-handles@1.x
- update to latest typescript/eslint config
- fix one usage of findDOMNode breaking StrictMode (didn't manage to fix the other one at react-transition-group)

#### How to test it?

https://modal--storecomponents.myvtex.com/?

### CODE REVIEW

File: react/components/TrapFocus.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ interface Props {
   open: boolean
 }
 
-const TrapFocus: React.FC<Props> = props => {
+const TrapFocus: React.FC<Props> = (props) => {
```
Comment (icazevedo) [inline react/components/TrapFocus.tsx:8]: Same here

**Review by Klynger — APPROVED** [review]

File: react/Modal.tsx [hunk 2]
```diff
@@ -90,6 +100,7 @@ function Modal(props: Props) {
 
   return (
     <BaseModal
+      classes={handles}
```
Comment (icazevedo) [inline react/Modal.tsx:103]: I think this prop should be renamed as `handles`

File: react/CloseButton.tsx [hunk 3]
```diff
@@ -1,32 +1,33 @@
...
 
-  const classes = {
+  const buttonClasses = {
     button: handles.closeButton,
     label: handles.closeButtonLabel,
   }
 
   return (
-    <Button classes={classes} onClick={handleClick}>
+    <Button classes={buttonClasses} onClick={handleClick}>
```
Comment (icazevedo) [inline react/CloseButton.tsx:30]: I think this `classes` prop should be renamed as `handles`.

File: react/components/Button.tsx [hunk 4]
```diff
@@ -8,15 +8,13 @@ interface AllClasses {
...
-type Classes = Partial<AllClasses>
-
 interface Props
   extends React.DetailedHTMLProps<
     React.ButtonHTMLAttributes<HTMLButtonElement>,
     HTMLButtonElement
   > {
   children: React.ReactNode
-  classes?: Classes
+  classes?: Partial<AllClasses>
```
Comment (icazevedo) [inline react/components/Button.tsx:17]: I think this prop should be renamed as `handles`.

File: react/components/Button.tsx [hunk 5]
```diff
@@ -8,15 +8,13 @@ interface AllClasses {
...
 }
 
 function Button(props: Props) {
```
Comment (icazevedo) [inline react/components/Button.tsx:20]: Same here (propsWithChildren)

**Review by icazevedo — CHANGES_REQUESTED** [review]

File: react/BaseModal.tsx [hunk 6]
```diff
@@ -144,6 +153,7 @@ export default function BaseModal(props: Props) {
     <Portal container={container}>
       <div
         {...rest}
+        className={`${props.classes.modal} ${props.className ?? ''}`}
```
[the remark this answers is not in the document]
Comment (igorbrasileiro) [reply react/BaseModal.tsx:156]: Also, can use `classes` instead of `props.classes`.

**Review by Klynger — APPROVED** [review]

File: react/components/Portal.tsx [hunk 7]
```diff
@@ -17,6 +18,10 @@ interface Props {
 export function getContainer(container?: ContainerType) {
   container = typeof container === 'function' ? container() : container
 
+  if (!container) {
+    return undefined
```
Comment (victorhmp) [inline react/components/Portal.tsx:22]: ⛏️  How about returning `null` instead of `undefined`?

Comment (victorhmp) [reply react/components/Portal.tsx:22]: Same thing at line 46

**Review by victorhmp — APPROVED** [review]

File: react/BaseModal.tsx [hunk 8]
```diff
@@ -20,6 +28,7 @@ interface Props
   disableEscapeKeyDown?: boolean
   children: React.ReactElement
   onBackdropClick?: (e: React.MouseEvent<HTMLDivElement>) => void
+  handles: CssHandlesTypes.CssHandles<typeof CSS_HANDLES>
```
Comment (igorbrasileiro) [inline react/BaseModal.tsx:31]: This component can be exported/imported if someone installs this app? If yes, this prop shouldn't be called `classes`? I don't have too much context

File: react/components/Backdrop.tsx [hunk 9]
```diff
@@ -1,23 +1,24 @@
...
   onClick?: (e: React.MouseEvent<HTMLDivElement>) => void
+  handles: CssHandlesTypes.CssHandles<typeof CSS_HANDLES>
 }
 
-export const CSS_HANDLES = ['backdropContainer', 'backdrop'] as const
-
-const Backdrop: React.FC<Props> = props => {
-  const { children, open, onClick, transitionDuration } = props
-  const handles = useCssHandles(CSS_HANDLES)
+const Backdrop: React.FC<Props> = (props) => {
```
Comment (igorbrasileiro) [inline react/components/Backdrop.tsx:19]: I saw that some functional components defined as arrow functions were converted to function. This isn't the same case?

```suggestion
function Backdrop(props: Props) {
```

File: react/components/TrapFocus.tsx (hunk 1 above)
Comment (igorbrasileiro) [reply react/components/TrapFocus.tsx:8]: Shouldn't convert to function? Same reason that i left before

**Review by igorbrasileiro — APPROVED** [review]
LGTM. I left some questions because i don't have context about it.

File: react/BaseModal.tsx (hunk 8 above)
Comment (klzns) [reply react/BaseModal.tsx:31]: Yes, it can, but it shouldn't 😅

Comment (klzns) [reply react/BaseModal.tsx:31]: Yes, it can, but it shouldn't 😅

File: react/components/Portal.tsx (hunk 7 above)
Comment (klzns) [reply react/components/Portal.tsx:22]: I can change here but not on line 46
image

File: react/components/Backdrop.tsx (hunk 9 above)
Comment (klzns) [reply react/components/Backdrop.tsx:19]: Changing!

**Review by icazevedo — APPROVED** [review]

**Pull request merged**